src/Lecture4.hs

```diff
@@ -225,8 +307,14 @@ Have a look at the 'sconcat' function from the 'Semigroup' typeclass to
 implement the next task.
 -}
 
+-- Couldn't use sconcat because Row is not an instance of Semigroup
+-- I guess we could implement <> for Row but it looks like we can do without
+--
+-- ... although I admit I arrived to this solution by trial and error, rather
+-- than reasoning ...
```

Comment:
You don't need an instance of `Semigroup` for `Row`. Usage of `fmap rowToStats` is the correct approach here 🙂 

You can't combine `Row`s but you can combine `Stats`. So, if you have `NonEmpty Row` you can transform it to `NonEmpty Stats` (using `fmap rowToStats` as you did) and then you can use `sconcat` 🙂 

Although, `foldl1' (<>)` is better than `sconcat` because it'll allow you to consume only constant amount of memory 👍🏻